cms/core/viewsets.py

```diff
@@ -47,3 +48,62 @@ class ContactDetailsViewSet(SnippetViewSet):
 
     index_view_class = ContactDetailsIndex
     chooser_viewset_class = ContactDetailsChooserViewset
+
+
+class GlossaryTermCreateView(SnippetCreateView):
+    def save_instance(self) -> GlossaryTerm:
+        """Automatically set the last updating user on Glossary Term on creation and modification."""
+        instance: GlossaryTerm = super().save_instance()
+        instance.updated_by = self.request.user
+        instance.save(update_fields=["updated_by"])
+        return instance
+
+
+class GlossaryTermEditView(SnippetEditView):
+    def save_instance(self) -> GlossaryTerm:
+        """Automatically set the last updating user on Glossary Term on creation and modification."""
+        instance: GlossaryTerm = super().save_instance()
+        instance.updated_by = self.request.user
+        instance.save(update_fields=["updated_by"])
+        return instance
+
+
+class GlossaryTermsIndex(SnippetIndexView):
+    list_display: ClassVar[list[str | Column]] = [
+        "name",
+        UpdatedAtColumn(),
+        UserColumn("updated_by"),
+        UserColumn("owner"),
+    ]
+
+
+class GlossaryTermsChooseColumnsMixin:
+    @property
+    def columns(self) -> list[Column]:
+        title_column = self.title_column  # type: ignore[attr-defined]
+        title_column.label = "Name"
+        return [title_column, UpdatedAtColumn(), UserColumn("updated_by")]
+
+
+class GlossaryChooseView(GlossaryTermsChooseColumnsMixin, SnippetChooseView): ...
+
+
+class GlossaryChooseResultsView(GlossaryTermsChooseColumnsMixin, SnippetChooseResultsView): ...
+
+
+class GlossaryChooserViewset(SnippetChooserViewSet):
+    choose_view_class = GlossaryChooseView
+    choose_results_view_class = GlossaryChooseResultsView
+
+
+class GlossaryViewSet(SnippetViewSet):
+    """A snippet viewset for Glossary."""
+
+    model = GlossaryTerm
+    icon = "list-ul"
+
+    index_view_class = GlossaryTermsIndex
+    chooser_viewset_class = GlossaryChooserViewset
+
+    add_view_class = GlossaryTermCreateView
+    edit_view_class = GlossaryTermEditView
```

Comment:
Minor, non-blocker: It would be nice to deduplicate selections. So users should not be able to have a definition selected more than once.